Fixes 17357: Use virtual chassis name as fallback for device

alehaa opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #17357

<!--
    Please include a summary of the proposed changes below.
-->

If a name is not defined for a device, it is automatically generated from other device data (i.e. virtual chassis member) if possible. This name is available as `label` attribute to not interfere with existing functionality or break APIs. This label will be used in several places like tables, while the API still returns the real name.

alehaa committed:
Index virtual chassis of device

In some use cases, a device does not have an individual name when it is
a member of a virtual chassis. By adding the virtual chassis to the
search index and making its name searchable, forms can use the virtual
chassis name instead in these situations.

alehaa committed:
Enhance display name of devices

If a name is not defined for a device, it is automatically generated
from other device data (i.e. virtual chassis member) if possible. This
name is available as 'label' attribute to not interfere with existing
functionality or break APIs.

arthanson commented on netbox/dcim/tests/test_models.py:
```diff
@@ -590,6 +590,34 @@ def test_device_duplicate_names(self):
+            virtual_chassis=virtual_chassis,
+            vc_position=2,
+        )
+        self.assertEqual(device3.label, 'VC 1:2')
```
Should have a test if set name and virtual_chassis that it returns the name, not the generated one.

alehaa committed:
Add test case for VC members with name

alehaa committed:
Don't cache device label

Generating the device label doesn't require much computing power. Thus,
caching does more harm than good, as it may return stale labels after
name changes. Now a fresh label is always generated.
